plugins/isp/internode_usage

```diff
@@ -68,46 +70,74 @@ fi
 
 xpath_extract() {
 	xpath="$1"
-	xpath -q -n -e "${xpath}" | \
-		sed 's/<\([^>]*\)>\([^<]*\)<[^>]*>/\2/'
+	# shellcheck disable=SC2064
+	trap "echo error extracting ${xpath} >&2" EXIT
+	xpath -q -n -e "${xpath}" \
+		| sed 's/<\([^>]*\)>\([^<]*\)<[^>]*>/\2/;s^N/A^U^'
+	trap - EXIT
 }
 
 xpath_extract_attribute() {
 	xpath="$1"
-	xpath -q -n -e "${xpath}" | \
-		sed 's/.*="\([^"]\+\)".*/\1/'
+	# shellcheck disable=SC2064
+	trap "error extracting attribute at ${xpath} >&2" EXIT
```

Comment:
How about using the `curl ... || echo >&2 "..."` approach instead?
(I think, it is easier to read here)

Another detail: I think, you omitted `echo` in the `trap`.

Reply:
Good catch on the missing `echo`.

I can't remember why I didn't just do `cmd || echo >&2`. I'll need to rethink that 😬

Reply:
Ah! Yes! That's it, you're right. I want to be able to output the error message, but to let the caller decide whether this is an error that can be tolerated (with `|| true`, which is used a few times).

So, yeah, I think we need to keep it like that.

Reply:
> Ah! Yes! That's it, you're right. [..]

A good example of code with surprising side effects :)

I would appreciate a more obvious approach, but I could live with this one, too.

Reply:
Ah! I think I found a better way. What about now?